### PR TITLE
Hack to get export default work

### LINKED ISSUE #1854: Typescript: 'export default function' gets replaced with 'export function'
[Reproduction](https://prettier.github.io/prettier/#%7B%22content%22%3A%22declare%20module%20%5C%22hello%5C%22%20%7B%5Cn%20%20%20%20export%20default%20function()%20%7B%7D%3B%5Cn%7D%22%2C%22options%22%3A%7B%22printWidth%22%3A80%2C%22tabWidth%22%3A2%2C%22singleQuote%22%3Afalse%2C%22trailingComma%22%3A%22none%22%2C%22bracketSpacing%22%3Atrue%2C%22jsxBracketSameLine%22%3Afalse%2C%22parser%22%3A%22typescript%22%2C%22semi%22%3Atrue%2C%22useTabs%22%3Afalse%2C%22doc%22%3Afalse%7D%7D)

Input:
```ts
declare module "hello" {
    export default function() {};
}
```

Output:

```ts
declare module "hello" {
  export function() {
  }
}
```
[Syntax error in output](https://prettier.github.io/prettier/#%7B%22content%22%3A%22declare%20module%20%5C%22hello%5C%22%20%7B%5Cn%20%20export%20function()%20%7B%5Cn%20%20%7D%5Cn%7D%22%2C%22options%22%3A%7B%22printWidth%22%3A80%2C%22tabWidth%22%3A2%2C%22singleQuote%22%3Afalse%2C%22trailingComma%22%3A%22none%22%2C%22bracketSpacing%22%3Atrue%2C%22jsxBracketSameLine%22%3Afalse%2C%22parser%22%3A%22typescript%22%2C%22semi%22%3Atrue%2C%22useTabs%22%3Afalse%2C%22doc%22%3Afalse%7D%7D)

### PR DESCRIPTION
**Pull request by vjeux**
Fixes #1854

### CODE REVIEW

**Comment by vjeux** [comment]
cc @azz

File: src/printer.js [hunk 1]
```diff
@@ -2399,7 +2399,15 @@ function genericPrintNoParens(path, options, print, args) {
       return concat(parts);
     case "TSNamespaceExportDeclaration":
       if (n.declaration) {
-        parts.push("export ", path.call(print, "declaration"));
+        // Temporary fix until https://github.com/eslint/typescript-eslint-parser/issues/263
+        const isDefault = options.originalText
+          .slice(util.locStart(n), util.locStart(n.declaration))
```
Comment (vjeux) [inline src/printer.js]: this is `"export default "` in this case.

**Comment by azz** [comment]
Cool. Are you tackling #1853 too? I have a branch checked out for it but I haven't made any progress yet.

**Comment by vjeux** [comment]
Feel free to do it, I need to do some more work on the release notes.

**Pull request merged**